File: ios/Alert.m
//
//  Text.m
//  BridgeNative
//
//  Created by Wahyu Setiawan on 20/10/21.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(MyAlert, NSObject)
RCT_EXTERN_METHOD(ShowMessage:(NSString *)message duration:(double *)duration)
@end

File: ios/Alert.swift
//
//  Text.swift
//  BridgeNative
//
//  Created by Wahyu Setiawan on 19/10/21.
//

import Foundation
import UIKit

@objc(MyAlert)

class MyAlert: NSObject, RCTBridgeModule{
    
  static func moduleName() -> String!{
    return "MyAlert";
  }
  
  @objc static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
  @objc func ShowMessage(_ message:NSString, duration:Double) -> Void {
    let alert = UIAlertController(title:nil, message: message as String, preferredStyle: .alert);
    let seconds:Double = duration;
    alert.view.backgroundColor = .white
    alert.view.layer.cornerRadius = 10
    
    DispatchQueue.main.async {
      (UIApplication.shared.delegate as? AppDelegate)?.window.rootViewController?.present(alert, animated: true, completion: nil);
    }
    
    DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + seconds, execute: {
      alert.dismiss(animated: true, completion: nil);
    })
  }
}
